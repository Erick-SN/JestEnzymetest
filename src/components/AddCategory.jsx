import React, { useState } from 'react';
import PropTypes from 'prop-types';

const AddCategory = ({ setCategories }) => {
  const [inputValue, setInputValue] = useState('');
  const handleSubmit = (e) => {
    e.preventDefault();
    if (inputValue.trim().length > 2) {
      setCategories((cats) => [inputValue, ...cats]);
      setInputValue('');
    }
  };

  // const handleChange = (e) => {
  //   setInputValue(e.target.value);
  //   console.log(e.target.value);
  // };

  // const isInputNumber = (e) => {
  //   const special = new RegExp('[0-9]');
  //   const ch = String.fromCharCode(!e.charCode ? e.which : e.charCode);
  //   if (!special.test(ch)) {
  //     e.preventDefault();
  //     return;
  //   }
  // };
  return (
    <form onSubmit={handleSubmit}>
      <p>{inputValue}</p>
      <input
        type='text'
        value={inputValue}
        onChange={(e) => setInputValue(e.target.value)}
        // onKeyPress={isInputNumber}
        // maxLength='10'
      />
    </form>
  );
};

AddCategory.propTypes = {
  setCategories: PropTypes.func.isRequired,
};

export default AddCategory;
