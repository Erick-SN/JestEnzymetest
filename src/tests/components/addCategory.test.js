import { shallow } from 'enzyme';
import React from 'react';
import '@testing-library/jest-dom';
import AddCategory from '../../components/AddCategory';

describe('Test at <AddCategory />', () => {
  const setCategories = jest.fn();
  let wrapper;
  beforeEach(() => {
    jest.clearAllMocks();
    wrapper = shallow(<AddCategory setCategories={setCategories} />);
  });
  test('Render component', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('should change inputValue', () => {
    const input = wrapper.find('input');
    const value = 'Hello world';
    input.simulate('change', { target: { value } });
    expect(wrapper.find('p').text().trim()).toBe(value);
  });

  test('should not execute the submit function', () => {
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).not.toHaveBeenCalled();
  });

  test('should call Setcategories and clear  input fields', () => {
    const value = 'Hello World';
    wrapper.find('input').simulate('change', { target: { value } });
    wrapper.find('form').simulate('submit', { preventDefault() {} });
    expect(setCategories).toHaveBeenCalled();
    expect(wrapper.find('input').prop('value')).toBe('');
  });
});
