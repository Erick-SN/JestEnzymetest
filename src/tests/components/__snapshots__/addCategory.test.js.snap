// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test at <AddCategory /> Render component 1`] = `ShallowWrapper {}`;
